bokeh/sphinxext/bokeh_example_sampledata.py

```diff
@@ -0,0 +1,149 @@
+from . import PARALLEL_SAFE
+from .bokeh_directive import BokehDirective
+from .util import get_sphinx_resources
+
+# -----------------------------------------------------------------------------
+# Globals and constants
+# -----------------------------------------------------------------------------
+
+__all__ = (
+    "BokehExampleSampledataDirective",
+    "setup",
+)
+
+RESOURCES = get_sphinx_resources()
+
+# -----------------------------------------------------------------------------
+# General API
+# -----------------------------------------------------------------------------
+
+# -----------------------------------------------------------------------------
+# Dev API
+# -----------------------------------------------------------------------------
+
+class sampledata_list(nodes.General, nodes.Element):
+    def __init__(self, *args, **kwargs):
+        self.sampledata_key = kwargs.pop("sampledata_key")
+        super().__init__(*args, **kwargs)
+
+class BokehExampleSampledataDirective(BokehDirective):
+    has_content = False
+    required_arguments = 1
+
+    def run(self):
+        return [sampledata_list('', sampledata_key=self.arguments[0])]
+
+
+def setup(app):
+    """ Required Sphinx extension setup function. """
+    app.add_node(sampledata_list)
+    app.add_directive("bokeh-example-sampledata", BokehExampleSampledataDirective)
+    app.connect('doctree-resolved', process_sampledata_xrefs)
+    app.connect('env-purge-doc', purge_xrefs)
+    app.connect('env-merge-info', merge_xrefs)
+    return PARALLEL_SAFE
+
+# -----------------------------------------------------------------------------
+# Private API
+# -----------------------------------------------------------------------------
+
+def purge_xrefs(app, env, docname):
+    if not hasattr(env, 'all_sampledata_xrefs'):
+        return
+
+    env.all_sampledata_xrefs = [xref for xref in env.all_sampledata_xrefs if xref['docname'] != docname]
+
+def merge_xrefs(app, env, docnames, other):
+    if not hasattr(env, 'all_sampledata_xrefs'):
+        env.all_sampledata_xrefs = []
+
+    if hasattr(other, 'all_sampledata_xrefs'):
+        env.all_sampledata_xrefs.extend(other.all_sampledata_xrefs)
+
+def process_sampledata_xrefs(app, doctree, fromdocname):
+
+    env = app.builder.env
+
+    if not hasattr(env, 'all_sampledata_xrefs'):
+        env.all_sampledata_xrefs = []
+
+    for node in doctree.traverse(sampledata_list):
+        sampladata_key = node.sampledata_key
+        content = []
+
+        this_sampladata_refs = []
+        for sample_info in env.all_sampledata_xrefs:
+            if sample_info['keyword'] == sampladata_key:
+                this_sampladata_refs.append(sample_info)
+
+        _len = len(this_sampladata_refs)
+        para = nodes.paragraph()
+        if _len:
+            s = '' if _len==1 else 's'
+            description = (
+            _(f'Check out the demonstration{s} for the ``{sampladata_key}`` sampledata. See the example{s} '))
```

Comment:
I think localizing would be great, I did set up the URL scheme to support different translations years ago e.g. `/en/latest` but the current status is that we don't use `_(...)` anywhere currently and I don't think we should just add it in one random place and nowhere else. This is something that should be considered and tackled as a coherent project. I'd ask to remove the call here and open a new issue to discuss the topic more broadly